synapse/handlers/presence.py

```diff
@@ -1982,6 +1982,14 @@ def handle_update(
                     obj=user_id,
                     then=new_state.last_active_ts + LAST_ACTIVE_GRANULARITY,
                 )
+            if prev_state.state == PresenceState.UNAVAILABLE:
+                # If we had idled out, but are still syncing, then new state will look
+                # like it's online. Verify with last_active_ts, as that is a sign of a
+                # pro-active event. Override that we are online when we probably aren't.
+                if now - new_state.last_active_ts > IDLE_TIMER:
+                    new_state = new_state.copy_and_replace(
+                        state=PresenceState.UNAVAILABLE
+                    )
```

Comment:
This feels a bit hacky -- can we avoid trying to update in this case at all instead backtracking from saying we're online to unavailable?